python/triqs/gf/tools.py

```diff
@@ -259,8 +259,94 @@ def fit_legendre(g_t, order=10):
     l = np.arange(len(lmesh))
     scale = np.sqrt(2.*l + 1) / mesh.beta
     scale = scale.reshape([len(lmesh)] + [1]*len(g_t.target_shape))
-    
+
     g_l = Gf(mesh=lmesh, target_shape=g_t.target_shape)
     g_l.data[:] = c_l.reshape(g_l.data.shape) / scale
 
     return g_l
+
+
+def discretize_bath(delta_inp, Nb , bandwidth = 3, t_val = 0.0, tol=1e-5):
+    """
+    discretizes a given Delta_iw with Nb bath sites
+
+    CAUTION: This function works only for 1x1 / scalar valued Gf objects.
+    if necessary call for each orbital in input Delta_iw separately.
+
+    Parameters
+    ----------
+    g : GfImFreq
+        Non-interacting Green's function.
+    Nb : int
+        number of bath sites
+    bandwidth : approximative bandwidth for init of bath energies
+    t_val : float
+        to which value should the bath hoppings be initialized
+    tol: float
+        tolerance for scipy minimize
+    Returns
+    -------
+    t_opt : list of optimized bath hopping parameters
+    e_opt : list of optimized bath energies
+    delta_iw : BlockGf (of GfImFreq) or GfImFreq
+               Hybridization function.
+    """
+    from scipy.optimize import minimize
+
+    ####
+    # define minimizer for scipy
+    def minimizer(parameters):
+        # first half of parameters are hoppings
+        t = parameters[0:Nb]
+        # scond half are bath energies
+        e = parameters[Nb:]
+
+        # build discretized bath function as
+        # delta = sum_nb t_n / (iw - eps_n)
+        for w in delta_disc.mesh:
+            delta_disc[w] = np.sum(t**2 / ( w.value - e ) ) - delta_inp[w]
+
+        # calculate norm sum_iw_n | delta_disc - delta_inp |^2
+        norm = (delta_disc*delta_disc.conjugate()).density().real
```

Comment:
I would not use density for the norm as this will trigger a tailfit of the possibly noisy data
```suggestion
        norm = np.linalg.norm(delta_disc.data - delta_iw.data)
```